infra/auth-server/src/utils/verify-token.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const jwt = require('jsonwebtoken')
+const logger = require('../logger')
+
+const { Sequelize, AuthTokenBlacklist } = require('../models')
+const { getbitAsync, setbitAsync, redisClient } = require('./redis')
+
+/**
+ * To check if token is blacklisted or revoked
+ *
+ * @param {String} authToken Token to verify
+ * @returns {Boolean} true if token is blacklisted
+ */
+const isBlacklisted = async authToken => {
+  const key = 'authtokens/' + authToken + '/revoked'
+  const revoked = await getbitAsync(key, 0)
+
+  if (revoked === 1) {
+    // Cache hit
+    return true
+  }
+
+  const monthAgo = Date.now() - 1000 * 60 * 60 * 24 * 30
+
+  const entry = await AuthTokenBlacklist.findOne({
+    where: {
+      authToken,
+      createdAt: {
+        // Only search the tokens blacklisted in the last month
```

Comment:
Is there a reason for this? Not immediately clear why tokens that are blacklisted aren't blacklisted forever. Maybe worth writing why only the last month is checked in the comment.

Reply:
As of now, we have an expiration time of 30 days. So irrespective of the status of the token, it will be invalid after 30 days from date of issuance. So, we don't have to run it against all the tokens in the DB. Kind of a short-circuit.